tests/rkt_image_export_test.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"crypto/sha512"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/ThomasRooney/gexpect"
+)
+
+const (
+	manifestExportTemplate = `{"acKind":"ImageManifest","acVersion":"0.6.1","name":"IMG_NAME","labels":[{"name":"version","value":"1.0.0"},{"name":"arch","value":"amd64"},{"name":"os","value":"linux"}],"app":{"exec":["/inspect"],"user":"0","group":"0","workingDirectory":"/","environment":[{"name":"VAR_FROM_MANIFEST","value":"manifest"}]}}`
+)
+
+func getHash(filePath string) (string, error) {
+	f, err := os.Open(filePath)
+	if err != nil {
+		return "", fmt.Errorf("error opening file: %v", err)
+	}
+
+	hash := sha512.New()
+	r := io.TeeReader(f, hash)
+
+	if _, err := io.Copy(ioutil.Discard, r); err != nil {
+		return "", fmt.Errorf("error reading file: %v", err)
+	}
+
+	return hex.EncodeToString(hash.Sum(nil)), nil
+}
+
+// TestImageExport tests 'rkt image export', it will import some existing
+// image, export it with rkt image export and check that the exported ACI hash
+// matches the hash of the imported ACI
+func TestImageExport(t *testing.T) {
+	testImage := "rkt-inspect-image-export.aci"
+	testImageName := "coreos.com/rkt-image-export-test"
+	expectManifest := strings.Replace(manifestExportTemplate, "IMG_NAME", testImageName, -1)
+
+	tmpDir, err := ioutil.TempDir("", "rkt-TestImageExport-")
+	if err != nil {
+		panic(fmt.Sprintf("Cannot create temp dir: %v", err))
+	}
+	defer os.RemoveAll(tmpDir)
+
+	tmpManifest, err := ioutil.TempFile(tmpDir, "manifest")
```

Comment:
TempFile() returns an open file and it should be closed with tmpManifest.Close().

But if you `defer tmpManifest.Close()`, would `tmpManifest.Name()` work in the following defer? I don't know if you need to copy the name in a new variable before closing the file.